core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -1355,47 +1351,14 @@ public void close() throws IOException {
     }
 
     @Override
-    public Text getFirstRow() throws IOException {
-      if (currentReaders.length == 0) {
-        return null;
```

Comment:
This old code would return null when the file was empty.  Now it returns something that explicitly means empty.  Other code seemed confused about what null meant and would expand it to an infinite range, which was harmless and resulted in files with no data going to all children tablets when a tablet split.  The new code returns some more specific than null to denote empty file.